src/Avalonia.Base/Input/Gestures.cs

```diff
@@ -102,11 +116,38 @@ private static void PointerPressed(RoutedEventArgs ev)
                 var e = (PointerPressedEventArgs)ev;
                 var visual = (Visual)ev.Source;
 
+                if(s_lastPointer != null)
+                {
+                    if(s_isHolding && ev.Source is Interactive i)
+                    {
+                        i.RaiseEvent(new HoldingRoutedEventArgs(HoldingState.Cancelled, s_lastPressPoint, s_lastPointer.Type));
+                    }
+                    s_holdCancellationToken?.Cancel();
+                    s_holdCancellationToken?.Dispose();
+                    s_holdCancellationToken = null;
+
+                    s_lastPointer = null;
+                }
+
+                s_isHolding = false;
+
                 if (e.ClickCount % 2 == 1)
                 {
                     s_isDoubleTapped = false;
                     s_lastPress.SetTarget(ev.Source);
+                    s_lastPointer = e.Pointer;
                     s_lastPressPoint = e.GetPosition((Visual)ev.Source);
+                    s_holdCancellationToken = new CancellationTokenSource();
+                    var token = s_holdCancellationToken.Token;
+                    var settings = AvaloniaLocator.Current.GetService<IPlatformSettings>();
+                    DispatcherTimer.RunOnce(() =>
+                    {
+                        if (!token.IsCancellationRequested && e.Source is InputElement i && i.IsHoldingEnabled && ( e.Pointer.Type != PointerType.Mouse || i.IsHoldWithMouseEnabled))
```

Comment:
Not really a suggestion, but just wonder if you can avoid creation of CancellationTokenSource on each pointer pressed by saving last pressed pointer.

Also, it should be decided if holding should work with multitouch in situation, when two pointers were pressed, both can get holding event. Not sure if we need it, but if we do - there should be a unit test for that as well.

Reply:
WinUI prevents holding from being triggered if there are more than 1 contact point. It would become a pinch gesture as soon as 2 contacts are detected.